controllers/dynakube/dynakube_controller.go

```diff
@@ -139,19 +140,29 @@ func (r *ReconcileDynaKube) Reconcile(ctx context.Context, request reconcile.Req
 	return reconcile.Result{RequeueAfter: rec.RequeueAfter}, nil
 }
 
-func (r *ReconcileDynaKube) reconcileImpl(ctx context.Context, rec *utils.Reconciliation) {
+func (r *ReconcileDynaKube) reconcileDynaKube(ctx context.Context, rec *utils.Reconciliation) {
 	dtcReconciler := DynatraceClientReconciler{
 		Client:              r.client,
 		DynatraceClientFunc: r.dtcBuildFunc,
 		UpdateAPIToken:      true,
 		UpdatePaaSToken:     true,
 	}
 	dtc, upd, err := dtcReconciler.Reconcile(ctx, rec.Instance)
+
 	rec.Update(upd, defaultUpdateInterval, "Token conditions updated")
 	if rec.Error(err) {
 		return
 	}
 
+	err = status.SetDynakubeStatus(rec.Instance, status.Options{
+		Dtc:       dtc,
+		ApiClient: r.apiReader,
+	})
+	if rec.Error(err) {
+		rec.Log.Error(err, "could not set Dynakube status")
+		return
```

Comment:
Do we really want to return here?
Wouldn't it be possible to create the AG or the OA even without having some of the status fields?

Reply:
It would be possible to create them without correctly set fields, but that would result in errors down the line. I thought it would be better to show errors early.